service/src/test/java/com/epam/esm/service/impl/CertificateServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.CertificateDao;
+import com.epam.esm.dao.parameters.ParametersProvider;
+import com.epam.esm.dao.parameters.SearchQueryProvider;
+import com.epam.esm.domain.dto.CertificateDTO;
+import com.epam.esm.domain.dto.TagDTO;
+import com.epam.esm.domain.entity.Certificate;
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.service.TagService;
+import com.epam.esm.service.exception.IdNotExistException;
+import com.epam.esm.service.mapper.CertificateDTOMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class CertificateServiceImplTest {
+
+    @Mock
+    private CertificateService certificateService;
+    @Mock
+    private CertificateDao certificateDao;
+    @Mock
+    private CertificateDTOMapper certificateDTOMapper;
+    @Mock
+    private TagService tagService;
+    @Mock
+    private SearchQueryProvider queryProvider;
+    @Mock
+    private ParametersProvider queryParametersProvider;
+
+    private static final String QUERY_PART = "query stub";
+    private static final String PARAMETER_STUB = "parameter stub";
+    private static final Integer RECORDS_AFFECTED_ONE = 1;
+    private static final Integer RECORDS_AFFECTED_NONE = 0;
+    private static final Long RECORD_ID = 1L;
+
+    @BeforeEach
+    public void setup() {
+        certificateService =
+                new CertificateServiceImpl(certificateDao, certificateDTOMapper, tagService, queryProvider, queryParametersProvider);

```

Comment:
This also could be performed By Spring/Test framework. Maybe @InjectMocks

Reply:
thanks for the hint. implemented